winlogbeat/beater/winlogbeat.go

```diff
@@ -175,6 +175,7 @@ func (eb *Winlogbeat) Stop() {
 	logp.Info("Stopping Winlogbeat")
 	if eb.done != nil {
 		close(eb.done)
+		eb.client.Close()
```

Comment:
I think I have some incorrect error handling [here](https://github.com/elastic/beats/pull/1402/files#diff-a5aa56cf8a91211ac65e7062080d439eR234).

(Making assumptions here because I haven't reviewed everything yet) When the publisher client closes some `PublishEvents()` calls might fail and return false. So I need to return from [here](https://github.com/elastic/beats/pull/1402/files#diff-a5aa56cf8a91211ac65e7062080d439eR243) to prevent any state from being persisted in the registry for the failed events. WDYT?